app/controllers/tags_controller.rb

```diff
@@ -21,6 +21,24 @@ def index
     @tags = params[:q].present? ? tags.search_by_name(params[:q]) : tags.order(hotness_score: :desc)
   end
 
+  def bulk
+    skip_authorization
+    @tags = Tag.includes(:badge).select(ATTRIBUTES_FOR_SERIALIZATION)
+
+    page = params[:page]
+    per_page = (params[:per_page] || 10).to_i
+    num = [per_page, 1000].min
+
+    if params[:tag_ids].present?
+      @tags = @tags.where(id: params[:tag_ids])
+    elsif params[:tag_names].present?
+      @tags = @tags.where(name: params[:tag_names])
+    end
```

Comment:
Looks like `tag_ids` and `tag_names` are mutually exclusive query params and `tag_ids` silently takes precedence if both are provided. Is this intentional?

Reply:
good observation and it is in fact intentional